### PR TITLE
[luci] Change a method of getting prefix in FuseBCQPass

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #4961

Until now, as each BCQ applicable candidate nodes are paired with their BCQ information nodes,
prefix and candidate nodes are paired as `std::map`.
However, because BCQ applicable candidate nodes are no longer paired and BCQ applicated weights are paired instead,
method of getting prefix is changed by comparing each weights values.
This commit will change the method.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseBCQPass.cpp [hunk 1]
```diff
@@ -25,6 +25,85 @@
...
 
+bool is_fusable_const(luci::CircleConst *before, luci::CircleConst *after, bool do_w_x)
+{
+  if (after->dtype() != loco::DataType::FLOAT32)
+    return false;
+
+  if (after->rank() != 2)
+    return false;
+
+  if (after->size<loco::DataType::FLOAT32>() != before->size<loco::DataType::FLOAT32>())
```
Comment (mhs4670go) [inline compiler/luci/pass/src/FuseBCQPass.cpp:36]: `before` should be checked if its `dtype` is `FLOAT32` as `after` does.

File: compiler/luci/pass/src/FuseBCQPass.cpp [hunk 2]
```diff
@@ -25,6 +25,85 @@
...
+  auto after_dim0 = after->dim(0).value();
+  auto after_dim1 = after->dim(1).value();
+
+  if (before->rank() == 2)
+  {
+    if (do_w_x)
+    {
+      // Check for [dim0, dim1] --> [dim0, dim1]
+      if (!(after->dim(0) == before->dim(0) && after->dim(1) == before->dim(1)))
+        return false;
```
Comment (mhs4670go) [inline compiler/luci/pass/src/FuseBCQPass.cpp:48]: This condition can be placed before this `if` block.

File: compiler/luci/pass/src/FuseBCQPass.cpp [hunk 3]
```diff
@@ -25,6 +25,85 @@
...
+
+      for (uint32_t i = 0; i < after->size<loco::DataType::FLOAT32>(); ++i)
+        if (after->at<loco::DataType::FLOAT32>(i) != before->at<loco::DataType::FLOAT32>(i))
+          return false;
+    }
+    else
+    {
+      // Check for [dim0, dim1] --> [dim1, dim0]
+      if (!(after->dim(0) == before->dim(1) && after->dim(1) == before->dim(0)))
+        return false;
```
Comment (mhs4670go) [inline compiler/luci/pass/src/FuseBCQPass.cpp:58]: with this.

File: compiler/luci/pass/src/FuseBCQPass.cpp (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/pass/src/FuseBCQPass.cpp:36]: This check is needed but not here, I will add this validation :)

File: compiler/luci/pass/src/FuseBCQPass.cpp (hunk 2 above)
Comment (llFreetimell) [reply compiler/luci/pass/src/FuseBCQPass.cpp:48]: Two condition is different...!

**Review by mhs4670go — DISMISSED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**